### ISSUE TITLE
Footable not working in "loaded content"

### ISSUE THREAD

**Issue by jahrralf** [fooplugins/FooTable#609]
Hi,

I have a page where I load html data and place it into a div. This data can also contain a script block with a document ready block, which then fires. Now I have a table in this loaded div which then gets initialized, converted into a footable.

With version 3.0.x it works, with version 3.1.4 I get the following message and the table is hidden/blocked by the loading animation of footable:

```
Uncaught TypeError: Cannot read property 'enabled' of null
    at footable.min.js:formatted:2918
    at jquery.min.js:2
    at i (jquery.min.js:2)
    at Object.add [as done] (jquery.min.js:2)
    at Array.<anonymous> (jquery.min.js:2)
    at Function.each (jquery.min.js:2)
    at Object.<anonymous> (jquery.min.js:2)
    at Function.a.Deferred (jquery-migrate.min.js:2)
    at Object.then (jquery.min.js:2)
    at k.<anonymous> (footable.min.js:formatted:2917)
```

It I load the page separately (so not in the div and so on), then everything works fine.

Thanks in advance,

Ralf

**Comment by steveush** [comment]
Hi,

It seems you are having an issue with the State component added in 3.1.0. I'm looking at the `:formatted` line number of 2918 in the minified file in Chrome and it seems to match the stack trace you provided. That said I'm not sure why you would be receiving this error, could you provide me with a jsfiddle demo'ing the issue? Or some simplified code that reproduces it so I can debug it on my end?

Thanks

**Comment by jahrralf** [comment]
Today I hate my life. So after a few hours checking trying experimenting I found out that if I remove the reference to vis.js, then everything works well. 

I just tried to reproduce this in a jsfiddle, however without the loading via Ajax, but did not see the issue there. http://jsfiddle.net/uy8rLuot/3/

So in the end I was staring here (line 6909) and looked if this.state is null or not. With vis.js, in my case, it was null, without it, is was defined:

```
	// hook into the _construct method so we can add the state property to the table.
	F.Table.extend('_construct', function(ready){
		this.state = this.use(FooTable.State);
		this._super(ready);
	});
```

How can this happen? What's going on? How to proceed in such a case? Any help is appreciated.

**Comment by steveush** [comment]
Seeing as you were debugging the variable I assume you did try using the compiled but **not** minified file (footable.js) instead of footable.min.js, correct? Just want to rule out the minification causing the issue.

**Comment by jahrralf** [comment]
Correct, I switched to the not minified version and had the same problems.

**Comment by steveush** [comment]
Ok, just so I can try make a mockup to replicate I just want to confirm I have the basic steps right.

1. Have both FooTable and vis included in the page.
2. Have an empty `div` to act as a container.
3. Ajax load the raw HTML content for the table that includes the `<script/>` tag that initializes FooTable.

From what you've said the script is included within a `jQuery.ready` function so the raw HTML being loaded and then appended to the container `div` should look something like the below.

```html
<script>
	jQuery(function($){
		$("#my-table").footable({/* Options */});
	});
</script>
<table id="my-table">
	<thead>
		<tr>
			<th data-type="number">ID</th>
			<th>Name</th>
		</tr>
	</thead>
	<tbody>
		<tr>
			<td>1</td>
			<td>Steve</td>
		</tr>
		<tr>
			<td>2</td>
			<td>Ralf</td>
		</tr>
	</tbody>
</table>
```

Is that right?

**Comment by jahrralf** [comment]
Yes, looks perfect. I am using the following block to initialize the table (both the script and the table are loaded via Ajax):

```
<script type="text/javascript">
    jQuery(document).ready(function ($) {
        // $(document).ready(function () {
        console.log("Document is ready...");

<? if ($sf_request->isXmlHttpRequest()): ?>
            // Popopver mit Ajax-Laden hinzufügen
            $('.ajaxPopover').popover({
                html: true,
                placement: "auto right",
                container: 'body',
                content: ajaxPopoverLoadData
            });
<? endif; ?>

        // FooTable initialisieren
        $('.my-footable-emails').footable();

        console.log("Document is ready... Finished.");
    });
</script>
```

**Comment by jahrralf** [comment]
I was not able to reproduce this in a jsfiddle. What I did to resolve this is that I wrapped the page with the vis.js part in an iframe. So I think we are all good as long as I am the only one with this issue...
